python_modules/dagster/dagster/_core/definitions/auto_materialize_rule.py

```diff
@@ -397,15 +514,20 @@ def evaluate_for_asset(self, context: RuleEvaluationContext) -> RuleEvaluationRe
             will_update_parents = will_update_parents_by_asset_partition[asset_partition]
 
             if updated_parents or will_update_parents:
-                conditions[
+                asset_partitions_by_evaluation_data[
                     ParentUpdatedRuleEvaluationData(
                         updated_asset_keys=frozenset(updated_parents),
                         will_update_asset_keys=frozenset(will_update_parents),
                     )
                 ].add(asset_partition)
-        if conditions:
-            return [(k, v) for k, v in conditions.items()]
-        return []
+
+        return self.add_evaluation_data_from_previous_tick(
+            context,
+            asset_partitions_by_evaluation_data,
+            should_use_past_data_fn=lambda ap: not context.materialized_requested_or_discarded_since_previous_tick(
```

Comment:
What if there's an in-progress run targeting the asset materialization which wasn't requested by auto-materialize? Does this PR change the handling in that case? Just checking.

Reply:
No change in behavior here -- `requested` only filters out runs requested by the asset daemon, which mirrors the current behavior.

Definitely possible that this is not the ideal behavior, but at least we're not changing it

Reply:
Cool